YourEasyRent/Controllers/ProductController.cs

```diff
@@ -17,77 +17,139 @@ public class ProductController : ControllerBase
         [HttpGet]
         public async Task<ActionResult<IEnumerable<Product>>> GetProducts()//   не надо бобавлять в скобках Id, потому что мы должне получить все продукты, поэтому и в операторах его нет Task<IEnumerable<Product>> GetProducts()
         {
-            var allProducts = await _repository.GetProducts();
+            try
+            {
+                var allProducts = await _repository.GetProducts();
+                return Ok(allProducts);// получаем все продукты
+            }
 
-            return Ok(allProducts); // получаем все продукты
+            catch (Exception ex) 
+            {
+                Console.WriteLine($"[GetProducts] : {ex.Message}");
+                return StatusCode(422, "Unable to process the request.");
 
+            }
         }
+
         [HttpGet("{id:length(24)}")]
         public async Task<ActionResult<Product>> Get(string id)
         {
-            var product = await _repository.Get(id);
-            if (product == null)
+            try
             {
-                return NotFound();
+                var product = await _repository.Get(id);
+                if (product == null)
+                {
+                    return NotFound();
+                }
+                return Ok(product);
+            }
+
+            catch (Exception ex) 
+            {
+                Console.WriteLine($"[Get] : {ex.Message}");
+                return StatusCode(404, "Unable to process the request. Product not found");
```

Comment:
404 is not appropriate here. As we will get an exception in case of a Null Reference, or mongoDB failure.